src/python/grpcio_observability/grpc_observability/observability_util.cc

```diff
@@ -0,0 +1,189 @@
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/python/grpcio_observability/grpc_observability/observability_util.h"
+#include "src/python/grpcio_observability/grpc_observability/server_call_tracer.h"
+#include "src/python/grpcio_observability/grpc_observability/client_call_tracer.h"
+
+#include "absl/strings/string_view.h"
+
+namespace grpc_observability {
+
+std::queue<CensusData>* kCensusDataBuffer;
```

Comment:
[The `k` prefix indicates that this is a constant](https://google.github.io/styleguide/cppguide.html#Constant_Names), which it is not (otherwise, the `const` keyword would show up somewhere). I think you want a [`g` prefix to indicate that it is a global](https://github.com/grpc/grpc/blob/eecc8978b653f426eebda3d2a40871302108ebeb/src/core/ext/transport/chttp2/transport/chttp2_transport.cc#L122). Also, [variable names in our version of C++ are generally in `snake_case`](https://google.github.io/styleguide/cppguide.html#Variable_Names). So, for example, this variable should be named `g_census_data_buffer`. Same for the accompanying mutex and condition variable.

I know these rules probably seem gratuitous, but they make the job of future readers of the code much easier. We don't have to pay much attention to variables that start with `k` (because they never change), but variables starting with `g` are suspicious because they can change over time and are often the root causes of bugs.

Reply:
Agree that we should follow naming conventions, and I'm trying to do so across all of the files. I must've missed those while checking...